### PR TITLE
bug fixes and more testing/coverage

### PR DESCRIPTION
**Pull request by tejashah88**
* added testing for request verification, HTTPS support, and POST-based routes
* fixed potential memory leaks from not closing the HTTPS server instance and not removing the hotswap listeners
* now using alexa-verifier-middleware for request verification
* changed loading location of contents of 'sslcert' folder (should be part of 'examples' folder)
* fixed documentation for generating the SSL certificate

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ### 2.3.2 (Next)
 
+* [#32](https://github.com/alexa-js/alexa-app-server/pull/32): group of bug fixes and more testing/coverage - [@tejashah88](https://github.com/tejashah88)
```
Comment (dblock) [inline CHANGELOG.md]: Lets expand this to be specific to be helpful to people reading the CHANGELOG, just list actual fixes on separate lines? Also start with a Capital letter and end with a period ;) `<nitpick />`

File: index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ var express = require('express');
 var alexa = require('alexa-app');
 var verifier = require('alexa-verifier');
 var bodyParser = require('body-parser');
+var avm = require('alexa-verifier-middleware');
```
Comment (dblock) [inline index.js]: I would call this `alexaVerifierMiddleware` to be consistent and remove the `var verifier` above that i think we no longer need?

File: index.js [hunk 3]
```diff
@@ -23,12 +24,16 @@ var appServer = function(config) {
     self.error = function(msg) { console.log(msg); };
 
     // Configure hotswap to watch for changes and swap out module code
-    hotswap.on('swap', function(filename) {
+    var swapCallback = function(filename) {
```
Comment (dblock) [inline index.js]: `hotswapCallback`? Minor.

File: package.json [hunk 4]
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "alexa-app": "^2.4.0",
     "alexa-verifier": "0.0.5",
```
Comment (dblock) [inline package.json]: We still need an explicit verifier dependency? Maybe not lock to a version and let avm bring it?

Comment (tejashah88) [inline package.json]: good catch for that!

**Comment by dblock** [comment]
Nice work, merged, thank you!

**Comment by dblock** [comment]
@tejashah88 Want to help maintaining this library? Make a release? I will add you to this repo and npm.

**Comment by tejashah88** [comment]
@dblock I would be more than willing to help maintain this library.

**Comment by dblock** [comment]
@tejashah88 You're in, you should have both repo and npm access. Maybe you can add something like https://github.com/alexa-js/alexa-app/blob/master/RELEASING.md and then make a release Please continue making pull requests for code, we code review each-others.

**Comment by tejashah88** [comment]
@dblock No worries, I have quite a few more PR's upcoming! For making the release, should I add the RELEASING.md, or should we make the release first?

**Comment by dblock** [comment]
I suggest copying/writing RELEASING and doing a release to "test" it. But really that's just my 0.02c, I want to make sure the next person releasing can just follow simple instructions.

**Comment by tejashah88** [comment]
Understood, I'll probably be doing some more bug fixing but I do have the RELEASING file ready.

**Pull request merged**